### PR TITLE
Implement the cake alias interface

### PR DESCRIPTION
**Pull request by miterosan**
This is for https://github.com/ppy/osu/pull/3135.

I tried to change as little as possible to make the original code compatible with the cake alias.

Please note that I moved the code from programm.cs to CodeSanityValidater.cs allowing the application to be used as command line app.

### CODE REVIEW

File: CodeFileSanity/CodeFileSanity.csproj [hunk 1]
```diff
@@ -1,12 +1,15 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup Label="Project">
-    <TargetFramework>netcoreapp2.1</TargetFramework>
+    <TargetFrameworks>netstandard2.0;net46</TargetFrameworks>
```
Comment (peppy) [inline CodeFileSanity/CodeFileSanity.csproj]: is the net46 necessary? feels like a step backwards.

**Comment by peppy** [comment]
Can we make `CodeSanityValidator` non-static and pass through `ValidateCodeSanitySettings` in all contexts (even from `Program.cs` running)?

Also you have public variables with lower-case naming, which isn't the standard formatting we use in projects.

File: CodeFileSanity/CodeFileSanity.csproj (hunk 1 above)
Comment (miterosan) [reply CodeFileSanity/CodeFileSanity.csproj]: Sadly addins require this framework combination. Cake automaticly targets net46 on windows and netstandard2.0 everywhere else. No clue why they choosed to do that that way.

**Comment by miterosan** [comment]
I agree. I just wanted to keep this pr small. Will do.

File: CodeFileSanity/CodeFileSanity.csproj (hunk 1 above)
Comment (peppy) [reply CodeFileSanity/CodeFileSanity.csproj]: what happens if net46 isn't present on windows? that could be a major issue.

Comment (miterosan) [reply CodeFileSanity/CodeFileSanity.csproj]: I'll check in a sec

Comment (miterosan) [reply CodeFileSanity/CodeFileSanity.csproj]: Ok I am back. it is possible to go netstandard2.0 only by using cake.coreclr.

**Comment by miterosan** [comment]
https://youtrack.jetbrains.com/issue/RSRP-410004

**Comment by peppy** [comment]
Oh, good to see they are actually looking at making that happen! Very promising.

File: CodeFileSanity/CodeSanityValidator.cs [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace CodeFileSanity
+{
+    public class CodeSanityValidator
+    {
+
```
Comment (peppy) [inline CodeFileSanity/CodeSanityValidator.cs]: newline

File: CodeFileSanity/CodeFileSanity.csproj [hunk 3]
```diff
@@ -1,12 +1,15 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup Label="Project">
-    <TargetFramework>netcoreapp2.1</TargetFramework>
+    <TargetFramework>netstandard2.0</TargetFramework>
```
Comment (peppy) [inline CodeFileSanity/CodeFileSanity.csproj]: Does this not mean that this project will no longer run as a non-library?

File: CodeFileSanity/CodeSanityValidator.cs [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+        public CodeSanityValidator(ValidateCodeSanitySettings settings) 
+        {
+            isAppveyorBuild = settings.IsAppveyorBuild;
+            rootDirectory = settings.RootDirectory;
+        }
+
+        public void Validate() 
+            => checkDirectory(rootDirectory);
+
+        string getLicenseHeader(string path)
```
Comment (peppy) [inline CodeFileSanity/CodeSanityValidator.cs]: missing `private` prefix

File: CodeFileSanity/CodeSanityValidator.cs [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+                if (s == "extensions: .cs")
+                {
+                    started = true;
+                }
+            }
+
+            return licenseHeader;
+        }
+
+        void checkDirectory(string path)
```
Comment (peppy) [inline CodeFileSanity/CodeSanityValidator.cs]: missing `private` prefix

File: CodeFileSanity/CodeSanityValidator.cs [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+            foreach (var sub in Directory.GetDirectories(path))
+                checkDirectory(sub);
+
+            var license = getLicenseHeader(path);
+
+            foreach (var file in Directory.GetFiles(path, "*.cs"))
+                checkFile(file, license);
+        }
+
+        void checkFile(string file, string licenseHeader)
```
Comment (peppy) [inline CodeFileSanity/CodeSanityValidator.cs]: missing `private` prefix

File: CodeFileSanity/CodeSanityValidator.cs [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+            if (Path.GetFileName(file) == "AssemblyInfo.cs")
+                return;
+
+            if (findMatchingLines(text, $"(enum|struct|class|interface) {Path.GetFileNameWithoutExtension(file).Split('_').First()}").Count == 0)
+            {
+                report(file, $"Filename does not match contained type.");
+            }
+        }
+
+        List<int> findMatchingLines(string input, string pattern, RegexOptions options = RegexOptions.None)
```
Comment (peppy) [inline CodeFileSanity/CodeSanityValidator.cs]: they all went missing. this wouldn't pass our normal CI.

**Review by peppy — CHANGES_REQUESTED** [review]

File: CodeFileSanity/CodeSanityValidator.cs (hunk 4 above)
Comment (miterosan) [reply CodeFileSanity/CodeSanityValidator.cs]: As far I know the default access modificator is private in c#.
I mean I have nothing against changing.

**Comment by miterosan** [comment]
any updates?

**Review by peppy — APPROVED** [review]
Looks to be working as expected. sorry for the delay.

**Pull request merged**